mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -825,6 +825,163 @@ def count_results_by_status(
+        filter_query = ""
+        now = mlrun.utils.datetime_now()
+        start = start or (now - timedelta(hours=24))
+        end = end or now
+
+        if application_names:
+            filter_query = self._generate_filter_query(
+                filter_column=mm_schemas.WriterEvent.APPLICATION_NAME,
+                filter_values=application_names,
+            )
+
+        def _get_latest_metrics_records(
+            type: Literal["metrics", "results"],
+        ) -> pd.DataFrame:
+            columns = [
+                mm_schemas.WriterEvent.END_INFER_TIME,
+                mm_schemas.WriterEvent.APPLICATION_NAME,
+            ]
+            if type == "results":
+                table = self.tables[
+                    mm_schemas.TDEngineSuperTables.APP_RESULTS
+                ].super_table
+                columns += [
+                    mm_schemas.ResultData.RESULT_NAME,
+                    mm_schemas.ResultData.RESULT_VALUE,
+                    mm_schemas.ResultData.RESULT_STATUS,
+                    mm_schemas.ResultData.RESULT_KIND,
+                ]
+                agg_column = mm_schemas.ResultData.RESULT_VALUE
+            else:
+                table = self.tables[mm_schemas.TDEngineSuperTables.METRICS].super_table
+                columns += [
+                    mm_schemas.MetricData.METRIC_NAME,
+                    mm_schemas.MetricData.METRIC_VALUE,
+                ]
+                agg_column = mm_schemas.MetricData.METRIC_VALUE
+
+            return self._get_records(
+                table=table,
+                start=start,
+                end=end,
+                columns=columns,
+                filter_query=filter_query,
+                timestamp_column=mm_schemas.WriterEvent.END_INFER_TIME,
+                group_by=columns[1:],  # without timestamp column
+                preform_agg_columns=[agg_column],
+                agg_funcs=["last"],
+            )
+
+        df_results = _get_latest_metrics_records(type="results")
+        df_metrics = _get_latest_metrics_records(type="metrics")
+
+        if df_results.empty and df_metrics.empty:
+            return metric_list
+
+        if not df_results.empty:
+            df_results.rename(
+                columns={
+                    f"last({mm_schemas.ResultData.RESULT_VALUE})": mm_schemas.ResultData.RESULT_VALUE,
+                },
+                inplace=True,
+            )
+            for _, row in df_results.iterrows():
+                metric_list.append(
+                    {
+                        "type": "result",
+                        "time": row[mm_schemas.WriterEvent.END_INFER_TIME],
+                        "name": row[mm_schemas.ResultData.RESULT_NAME],
+                        "kind": row[mm_schemas.ResultData.RESULT_KIND],
+                        "status": row[mm_schemas.ResultData.RESULT_STATUS],
+                        "value": row[mm_schemas.ResultData.RESULT_VALUE],
+                    }
```

Comment:
The metric/result dicts returned have an implicit schema, maybe we can consider defining a pydantic model or type alias to make the structure explicit?